### PR TITLE
[WIP] Added functionality to build index from file

### PR DESCRIPTION
**Pull request by grubdragon**
* Adds a SELECT index_from_external as required
* Adds function to do the actual build from a 
  binary at "path"

Pending
* Verification of functions being written
  * Is this how we want to structure it?
* Integration between functionality and SELECT statement
   * Confirm and understand workflow of SELECT - at what point is it called? How to test it?
* Testing

### CODE REVIEW

File: src/hnsw.c [hunk 1]
```diff
@@ -229,3 +229,20 @@ Datum       vector_l2sq_dist(PG_FUNCTION_ARGS)
...
+
+PGDLLEXPORT PG_FUNCTION_INFO_V1(index_from_external);
+Datum       index_from_external(PG_FUNCTION_ARGS)
+{
+    char *tablename_str = PG_GETARG_CSTRING(0);
+    char *index_path_str = PG_GETARG_CSTRING(1);
+
+    // TODO : Call the build function
+    bool index_created = false;
+    // bool index_created = build_index(tablename_str, index_path_str);
```
Comment (Ngalstyan4) [inline src/hnsw.c:241]: You may need to call [`index_create`](https://github.com/postgres/postgres/blob/3f8c98d0b631882e3c513954df42891c1c9585c7/src/backend/catalog/index.c#L666-L732) from here.

File: src/hnsw.c [hunk 2]
```diff
@@ -229,3 +229,20 @@ Datum       vector_l2sq_dist(PG_FUNCTION_ARGS)
...
+
+    // Free the allocated memory for converted C strings
+    pfree(tablename_str);
```
[the remark this answers is not in the document]
Comment (grubdragon) [reply src/hnsw.c]: Removed

File: src/hnsw/build.h [hunk 3]
```diff
@@ -39,7 +39,9 @@ typedef enum
 } HnswDataType;
 
 IndexBuildResult *ldb_ambuild(Relation heap, Relation index, IndexInfo *indexInfo);
+IndexBuildResult *ldb_ambuild_from_file(Relation heap, Relation index, IndexInfo *indexInfo, char const *path);
 void              ldb_ambuildunlogged(Relation index);
+void              ldb_ambuildunlogged_from_file(Relation index, char const *path);
```
[the remark this answers is not in the document]
Comment (grubdragon) [reply src/hnsw/build.h]: Addressed this comment, resolving!

File: src/hnsw/build.c [hunk 4]
```diff
@@ -297,6 +357,23 @@ IndexBuildResult *ldb_ambuild(Relation heap, Relation index, IndexInfo *indexInf
     return result;
 }
 
+/*
+ * Build the index for a logged table from a usearch bin file
+ */
+IndexBuildResult *ldb_ambuild_from_file(Relation heap, Relation index, IndexInfo *indexInfo, char const *path)
```
[the remark this answers is not in the document]
Comment (grubdragon) [reply src/hnsw/build.c]: Removed

File: src/hnsw.c [hunk 5]
```diff
@@ -229,3 +229,20 @@ Datum       vector_l2sq_dist(PG_FUNCTION_ARGS)
...
+    pfree(index_path_str);
+
```
[the remark this answers is not in the document]
Comment (grubdragon) [reply src/hnsw.c:408]: This works now, resolving

File: src/hnsw.c [hunk 6]
```diff
@@ -195,6 +209,180 @@ static float8 vector_dist(Vector *a, Vector *b, usearch_metric_kind_t metric_kin
...
+    Oid  tableSpaceId = get_rel_tablespace(table_id);
+    if(tableSpaceId == InvalidOid) {
+        tableSpaceId = GetDefaultTablespace(heapRelation->rd_rel->relpersistence, partitioned);
+    }
+
+    elog(INFO, "tableSpaceId: %u", tableSpaceId);
+
+    // Set the Oids for collations and ops
+    Oid   collationObjectId[] = {InvalidOid};
+    Oid   classObjectId[] = {InvalidOid};
```
Comment (grubdragon) [inline src/hnsw.c:346]: @Ngalstyan4 I have a question on this as well, not sure what goes here. I have tried to understand what these do to no avail

**Comment by grubdragon** [comment]
After multiple iterations of this code and some offline discussion, it was determined that this is not the way to go, hence made sense to make a fresh PR instead. Closing.

**Pull request closed without merging**